plugins/bulk-import/src/components/Repositories/CatalogInfoAction.tsx

```diff
@@ -123,9 +121,9 @@ const CatalogInfoAction = ({ data }: { data: AddRepositoryData }) => {
           <IconButton
             target="_blank"
             href={
-              values?.repositories[data.id]?.catalogInfoYaml?.prTemplate
+              values?.repositories?.[data.id]?.catalogInfoYaml?.prTemplate
                 ?.pullRequestUrl ||
-              values?.repositories[data.id]?.repoUrl ||
+              values?.repositories?.[data.id]?.repoUrl ||
```

Comment:
In this case we have a button that doesn't work right? Can we extract that code (`const link = ` and then when there is no link add an disabled button with a tooltip?

```tsx
if (!link) {
  return (
    <Tooltip title="Link is missing (or whatever)">
      <IconButton ... disabled />
    <Tooltip>
  )
}
```

Reply:
The edit and open in new tab icons are displayed based on the user's permissions, depending on whether they can edit the pull request

Reply:
Anyway, the code generates a link that isn't clickable in some cases.

With the recommended check we can ensure that we don't create a button that doesn't open a valid link.